src/model/TaskManager.cpp

```diff
@@ -48,76 +84,85 @@ std::vector<Core::TaskEntity> TaskManager::getTasks() const {
 std::vector<Core::TaskEntity> TaskManager::getTasks(const std::string &label) const {
     std::vector<Core::TaskEntity> tasks;
     for (const auto &kv : tasks_) {
-        if (kv.second.first.label() == label) {
-            Core::TaskEntity te;
-            te.set_allocated_id(new Core::TaskID(kv.first));
-            te.set_allocated_data(new Core::Task(kv.second.first));
-            tasks.push_back(te);
+        for (const auto &task_label : kv.second.first.labels()) {
+            if (task_label == label) {
+                Core::TaskEntity te;
+                te.set_allocated_id(new Core::TaskID(kv.first));
+                te.set_allocated_data(new Core::Task(kv.second.first));
+                tasks.push_back(te);
+                break;
+            }
         }
     }
     return tasks;
 }
 
-std::vector<Core::TaskEntity> TaskManager::getTasks(const Core::TaskID &id) const {
+std::vector<Core::TaskEntity> TaskManager::getTaskWithSubtasks(const Core::TaskID &id) const {
     std::vector<Core::TaskEntity> tasks;
     Core::TaskEntity te;
     te.set_allocated_id(new Core::TaskID(id));
-    te.set_allocated_data(new Core::Task(tasks_.at(id).first));
+    std::optional<Core::Task> task = GetTask(id);
+    if (task)
+        te.set_allocated_data(new Core::Task(*task));
     // not including parent, but will include children
     tasks.push_back(te);
-    for (const auto &ch_id : tasks_.at(id).second.children()) {
-        auto ch_tasks = getTasks(ch_id);
+    for (const auto &ch_id : ChildrenOf(id)) {
+        auto ch_tasks = getTaskWithSubtasks(ch_id);
         for (auto &ch_task : ch_tasks) {
-            ch_task.set_allocated_parent(new Core::TaskID(*tasks_.at(ch_task.id()).second.parent()));
+            ch_task.set_allocated_parent(new Core::TaskID(*ParentOf(ch_task.id())));
         }
         tasks.insert(tasks.end(), ch_tasks.begin(), ch_tasks.end());
     }
     return tasks;
 }
 
 ActionResult TaskManager::Delete(const Core::TaskID &id, bool deleteChildren) {
-    try {
-        if (!tasks_.at(id).second.children().empty() && !deleteChildren)
+    if (Validate(id)) {
```

Comment:
you don't "Validate" `id`, you're checking if it exists

Reply:
renamed to `IsPresent`